File: Grind75/Week 1/problem121.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        left = 0
        right = 1
        max_profit = 0
        while right < len(prices):
            profit = prices[right] - prices[left]
            if profit > 0:
                max_profit = max(profit, max_profit)
            else:
                left = right
            right += 1
        return max_profit

File: problem121.cpp
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        if (prices.size() == 1)
            return 0;
        int profit = prices[1] - prices[0];
        int mins = prices[0];
        for (int i = 0; i < prices.size() - 1; i++)
        {
            profit = max(profit, prices[i] - mins);
            mins = min(prices[i], mins);
        }
        profit = max(prices[prices.size() - 1] - mins, profit);
        return profit;
    }
};